lib/worker.js

```diff
@@ -150,6 +155,23 @@ async function updateDependency(upgrade) {
       newVersion);
   }
 
+  function getYarnLockFileName(packageFileName) {
+    const packagePath = packageFileName.split('/');
+    // Remove 'package.json'
+    const basePath = packagePath.slice(0, -1);
+    basePath.push('yarn.lock');
+    return basePath.join('/');
+  }
+
+  function generateLockFile(newPackageJson) {
+    logger.debug('Generating new yarn.lock file');
+    rimraf.sync('/tmp/renovate-yarn');
+    mkdirp.sync('/tmp/renovate-yarn');
```

Comment:
This may lead to conflicts on shared Linux boxes (where two or more users are logged into a Linux server, and running `renovate` through `cron`, or manually).

In these situations I typically fallback on `tmp` - https://www.npmjs.com/package/tmp#synchronous-directory-creation

It has support for synchronous directory creation (with unique name), and a cleanup callback function.